Simplenote/CoreDataManager.swift

```diff
@@ -9,27 +15,44 @@ class CoreDataManager: NSObject {
     private(set) var managedObjectContext: NSManagedObjectContext!
     private(set) var persistentStoreCoordinator: NSPersistentStoreCoordinator!
 
-    init(_ storageURL: URL, storageSettings: StorageSettings = StorageSettings()) throws {
+    init(_ storageURL: URL, storageSettings: StorageSettings = StorageSettings(), for usageType: CoreDataUsageType = .standard) throws {
         super.init()
+        managedObjectModel = NSManagedObjectModel(contentsOf: storageSettings.modelURL)!
+        managedObjectContext = objectContext()
+        persistentStoreCoordinator = try storeCoordinator(with: managedObjectModel, storageURL: storageURL)
 
-        guard let mom = NSManagedObjectModel(contentsOf: storageSettings.modelURL) else {
-            fatalError("Cannot load model")
-        }
-        managedObjectModel = mom
+        setupStackIfNeeded(mainContext: managedObjectContext, psc: persistentStoreCoordinator, for: usageType)
+    }
 
-        managedObjectContext = NSManagedObjectContext(concurrencyType: .mainQueueConcurrencyType)
-        managedObjectContext.undoManager = nil
+    private func objectContext() -> NSManagedObjectContext {
```

Comment:
We should prob. rename `objectContext()`, since its purpose is not clear: it's building and returning a new instance. `objectContext` would be proper for a property name.

How abooouttttt:

- `buildMainContext`
- `newMainContext`
- `createMainContext`

?